### PR TITLE
fix(57451): Prevent self-imports when using the "Move to File" refactor

### LINKED ISSUE #57451: Incorrect import destination after using `Move to file`
Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: 1.87.0-insider (Universal)
- OS Version: macOS 14.3.1

Steps to Reproduce:

1. Create files `utils.ts` and `program.ts`
2. Create and export any function inside `utils.ts`.
3. Import this function from `program.ts`.
4. Move the function into `program.ts` using the `Refactor` -> `Mote to file` action.

**Expected result**:
The function should be excluded from the import in `program.ts`.

**Actual result**:
The import statement for the function remains present, and the import destination changes from `utils` to `program` (which would not work).

### PR DESCRIPTION
**Pull request by eloyrobillard**
Fixes #57451

### CODE REVIEW

**Comment by eloyrobillard** [comment]
@microsoft-github-policy-service agree

File: src/services/refactors/moveToFile.ts [hunk 1]
```diff
@@ -430,6 +430,10 @@ export function updateImportsInOtherFiles(
                 deleteUnusedImports(sourceFile, importNode, changes, shouldMove); // These will be changed to imports from the new file
 
                 const pathToTargetFileWithExtension = resolvePath(getDirectoryPath(oldFile.path), targetFileName);
+
+                // no self-imports
+                if (sourceFile.fileName === pathToTargetFileWithExtension) return;
+
```
Comment (sheetalkamat) [inline src/services/refactors/moveToFile.ts:436]:
```suggestion
                const pathToTargetFileWithExtension = resolvePath(getDirectoryPath(getNormalizedAbsolutePath(oldFile.fileName, program.getCurrentDirectory())), targetFileName);
                // no self imports
                if (getStringComparer(!program.useCaseSensitiveFileNames())(pathToTargetFileWithExtension, sourceFile.fileName) === Comparison.EqualTo) return; 

```

Comment (sheetalkamat) [reply src/services/refactors/moveToFile.ts:436]: @navya9singh  we also need to replace usage of `sourceFile.path` in `makeImportOrReqiure` so its not using canonical path but actual file name: 
```ts
 const pathToTargetFile = resolvePath(getDirectoryPath(sourceFile.path), targetFileNameWithExtension);
```
to 
```ts
const pathToTargetFile = resolvePath(getDirectoryPath(getNormalizedAbsolutePath(sourceFile.fileName, program.getCurrentDirectory())), targetFileNameWithExtension);

```

**Review by sheetalkamat — CHANGES_REQUESTED** [review]

File: src/services/refactors/moveToFile.ts (hunk 1 above)
Comment (eloyrobillard) [reply src/services/refactors/moveToFile.ts]: made the changes

**Review by sheetalkamat — APPROVED** [review]

**Comment by navya9singh** [comment]
Thanks for working on this! This pr also fixes https://github.com/microsoft/TypeScript/issues/54309

**Review by navya9singh — APPROVED** [review]

**Pull request merged**